src/app/pages/arbol-cuentas-contables/selector-contable/selector-contable.component.ts

```diff
@@ -0,0 +1,180 @@
+        if (index === 0) {
+          this.formValue[this.normaContable[index].key] = this.cuentasList.find((element) => {
+            element.Nombre = `${element.data.Codigo} ${element.data.Nombre}`;
+            return element.data.Codigo === codigo;
+          });
+        } else {
+          this.formValue[this.normaContable[index].key] = this.formValue[this.normaContable[index].parent].children.find(
+            (element) => {
+
+              element.Nombre = `${element.data.Codigo} ${element.data.Nombre}`;
+              return element.data.Codigo === codigos.slice(0, index + 1).join('-');
+            }
+          );
+        }
+      });
+    }
+  }
+
+  buildForm() {
+    this.selectorForm = {
+      controls: this.normaContable.slice(0, this.claseMinima).map((item, index) => {
+        const control = this.createItemSelector(item);
+        const formControl = new FormControl(this.formValue[item.key]);
+        if (index === 0) {
+          (control as any).optionList.elements = () =>
+            this.cuentasList.map((element) => ({
+              ...element,
+              Nombre: `${element.data.Codigo} ${element.data.Nombre}`,
+            }));
+        }
+        this.form.addControl(control.key, formControl);
+        return control;
+      }),
+    };
+  }
+
+  addControl() {
+    const norma = this.normaContable[this.selectorForm.controls.length];
+    if (norma) {
+      const control = this.createItemSelector(norma);
+      this.selectorForm = {
+        controls: [
+          ...this.selectorForm.controls,
+          control
+        ]
+      };
+      this.form.addControl(control.key, new FormControl(''));
+      this.form.controls[norma.parent].valueChanges.subscribe((value) => {
+        if (value) {
+          this.form.get(norma.key).setValue('');
+        }
+      });
+    }
+  }
+
+  deleteControl() {
+    const norma = this.normaContable[this.selectorForm.controls.length - 1];
+    if (norma && this.selectorForm.controls.length > this.claseMinima) {
+      this.selectorForm.controls.pop();
+      this.form.removeControl(norma.key);
+    }
+  }
+
+  createItemSelector(item): (ReactiveFormGroup | ReactiveFormControl) {
+    return {
+      key: item.key,
+      type: 'autocomplete',
+      label: item.label,
+      defaultValue: '',
+      placeholder: item.label,
+      validators: [CustomValidators.customRequired('Campo Requerido')],
+      optionList: {
+        elements: (parent) =>
+          parent &&
+          parent.value[item.parent] &&
+          parent.value[item.parent].children ?
+          parent.value[item.parent].children.map(
+            (element) => ({
+              ...element,
+              Nombre: `${element.data.Codigo} ${element.data.Nombre}`,
+            })
+          ) : [],
+        labelKey: 'Nombre',
+      },
+      valueChanges: (parent) => {
+        parent.get(item.child) && typeof parent.get(item.key).value !== 'object' ? parent.get(item.child).setValue('') : null;
```

Comment:
Si no se requiere el valor retornado por una terna, optar por una alternativa explícita
```suggestion
        if (parent.get(item.child) && typeof parent.get(item.key).value !== 'object') {
          parent.get(item.child).setValue('');
        }
```
**NOTA:** Revisar si este _suggested change_ tal cual como se dejó, funcionaría o si habría que ajustarlo o si no aplica...